common/src/dir_utils.rs

```diff
@@ -53,3 +54,68 @@ pub fn default_path(filename: &str, base_path: Option<&PathBuf>) -> PathBuf {
     home.push(filename);
     home
 }
+
+pub fn absolute_path<P>(path: P) -> PathBuf
+where P: AsRef<std::path::Path> {
+    let path = path.as_ref();
+    let absolute_path = if path.is_absolute() {
+        path.to_path_buf()
+    } else {
+        std::env::current_dir().unwrap_or_default().join(path)
+    }
+    .clean();
+    absolute_path
+}
+
+#[cfg(test)]
+mod test {
+    use crate::dir_utils;
+    use std::path::PathBuf;
+    use tari_test_utils::random::string;
+    use tempdir::TempDir;
+
+    #[test]
+    fn test_multiple_levels_create_data_directory() {
+        let temp_dir = TempDir::new(string(8).as_str()).unwrap();
+        let dir = &PathBuf::from(
+            temp_dir.path().to_path_buf().display().to_string().to_owned() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/" +
+                string(2).as_str() +
+                "/",
+        );
```

Comment:
You could do this in one line with something like
`[0..12].map(|_| string(2)).collect().join("/")`